React/Views/RCTView.m

```diff
@@ -184,6 +184,53 @@ - (BOOL)didActivateAccessibilityCustomAction:(UIAccessibilityCustomAction *)acti
   return YES;
 }
 
+- (NSString *)accessibilityValue
+{
+  NSMutableString *value = [NSMutableString stringWithString:@""];
+  NSDictionary *roleDescriptions = @{
+                                     @"alert" : @"alert",
+                                     @"checkbox" : @"checkbox",
+                                     @"combobox" : @"combo box",
+                                     @"menu" : @"menu",
+                                     @"menubar" : @"menu bar",
+                                     @"menuitem" : @"menu item",
+                                     @"progressbar" : @"progress bar",
+                                     @"radio" : @"radio button",
+                                     @"radiogroup" : @"radio group",
+                                     @"scrollbar" : @"scroll bar",
+                                     @"spinbutton" : @"spin button",
+                                     @"switch" : @"switch",
+                                     @"tab" : @"tab",
+                                     @"tablist" : @"tab list",
+                                     @"timer" : @"timer",
+                                     @"toolbar" : @"tool bar",
+                                     };
+  NSDictionary *stateDescriptions = @{
+                                      @"on" : @"on",
+                                      @"off" : @"off",
+                                      @"checked" : @"checked",
+                                      @"unchecked" : @"not checked",
+                                      @"busy" : @"busy",
+                                      @"expanded" : @"expanded",
+                                      @"collapsed" : @"collapsed",
+                                      };
+  NSString *roleDescription = _accessibilityRole ? roleDescriptions[_accessibilityRole]: nil;
+  if (roleDescription) {
+    [value appendString:roleDescription];
+  }
+  for (NSString *state in _accessibilityStates) {
+    NSString *stateDescription =state ? stateDescriptions[state] : nil;
+    if (stateDescription) {
+        [value appendString:@" "];
```

Comment:
separator here should be a "," since VoiceOver will make it sound more natural when reading a list of words.

Reply:
Inserted comment in the wrong field... I made this a space due to eventual localization concerns (this whole chunk of code needs a rethink to support localizing role and state descriptions). Seems like the right separator is likely different depending on language.

Reply:
The separator here doesn't need to be localized, it should be a comma regardless of localization. VoiceOver will insert pauses when it encounters commas, which will help this list of states sound more natural. The list of localized states should be comma separated in any translation. Otherwise, this list of states would be space separated and VoiceOver may read them out like a sentence.

e.g. the final string should be "combobox, radio, alert" instead of "combobox radio alert". Even when localized, it would be "<state A>, <state B>, <state C>"